bert/glue.py

```diff
@@ -0,0 +1,391 @@
+            num_labels=num_labels,
+            pretrained_model_name=cfg.pretrained_model_name,
+            pretrained_checkpoint=cfg.get('pretrained_checkpoint', None),
+            model_config=cfg.get('model_config', None),
+            tokenizer_name=cfg.get('tokenizer_name', None),
+            gradient_checkpointing=cfg.get('gradient_checkpointing', None)
+        )
+    else:
+        raise ValueError(f'Not sure how to build model with name={cfg.name}')
+
+
+def get_values_from_path(path: str, separator: str = '/') -> Dict[str, str]:
+    """Parses out information from a path/string that looks like ...<separator>key=value<separator..."""
+    dict_output = {}
+    underscore_split = path.split(separator)
+    for item in underscore_split:
+        if '=' not in item:
+            continue
+
+        key, value = item.split('=')
+        dict_output[key] = value
+    return dict_output
+
+
+def get_checkpoint_name_from_path(path: str) -> str:
+    """To go from checkpoint name to path, replace | with /"""
+    return path.lstrip('/').replace('/', '|')
+
+
+def download_starting_checkpoint(starting_checkpoint_load_path: str,
+                                 local_pretrain_checkpoints_folder: str) -> str:
+    """Downloads the pretrained checkpoints to start from. Currently only supports S3 and URLs"""
+    load_object_store = None
+    parsed_path = urlparse(starting_checkpoint_load_path)
+    if parsed_path.scheme == "s3":
+        load_object_store = S3ObjectStore(bucket=parsed_path.netloc)
+
+    download_path = parsed_path.path if parsed_path.scheme == "s3" else starting_checkpoint_load_path
+    os.makedirs(local_pretrain_checkpoints_folder, exist_ok=True)
+    local_path = os.path.join(local_pretrain_checkpoints_folder, get_checkpoint_name_from_path(parsed_path.path))
+    if not os.path.exists(local_path):
+        get_file(destination=local_path, path=download_path.lstrip('/'), object_store=load_object_store, progress_bar=True)
+
+    return local_path
+
+
+def _setup_gpu_queue(num_gpus: int, manager: SyncManager):
+    """Returns a queue with [0, 1, .. num_gpus]."""
+    gpu_queue = manager.Queue(num_gpus)
+    for gpu_id in range(num_gpus):
+        gpu_queue.put(gpu_id)
+    return gpu_queue
+
+
+def create_job_configs(main_config: om.DictConfig, tasks_to_run: Set[str], pretrained_checkpoint_path: str):
+    configs = []
+    for task_name, task_config in main_config.tasks.items():
+        if task_name not in tasks_to_run:
+            continue
+        for task_seed in task_config.get('seeds', [main_config.default_seed]):
+            run_name = f'{main_config.base_run_name}_task={task_name}_seed={str(task_seed)}'
+            logger_configs = copy.deepcopy(main_config.get('loggers', {}))
+            for logger_name, logger_config in logger_configs.items():
+                if logger_name == 'wandb':
+                    logger_config['group'] = main_config.base_run_name
+                    logger_config['name'] = run_name
```

Comment:
Adding small suggestions here for myself.
* It could be helpful to add the following line for plotting and filtering purposes in w&b:
```logger_config['init_kwargs'] = {'config': {'task': task_name, 'seed': task_seed}}```
* It could be helpful to allow use to define `group` in the config; if user doesn't specify, then we can set it automatically to the base_run_name

Reply:
I'm going to hold off on this for now until I can verify that it works for my own pipeline